### PR TITLE
fix fix/MUWM-4347

### PR DESCRIPTION
**Pull request by fanglinfang**
Built on my-dev
https://jira.cac.washington.edu/browse/MUWM-4347

Also fixed time zone RuntimeWarnings.

### CODE REVIEW

File: myuw/dao/myuw_notice.py [hunk 1]
```diff
@@ -1,60 +1,65 @@
...
             user_notices.append(notice)
             continue
+
         if notice.is_past_student and affiliations["past_stud"]:
             user_notices.append(notice)
             continue
+
         for key in affiliations:
+            # exclude campuses
+            if key == 'seattle' or key == 'bothell' or key == 'tacoma':
```
Comment (fanglinfang) [inline myuw/dao/myuw_notice.py]: This is the cause #2.

File: myuw/dao/myuw_notice.py [hunk 2]
```diff
@@ -1,60 +1,65 @@
...
+                notice.is_grad or notice.is_grad_c2 or
+                notice.is_pce or notice.is_student or
                 notice.is_undergrad or notice.is_undergrad_c2 or
                 notice.is_fyp or notice.is_past_student or
                 notice.is_clinician or notice.is_employee or
                 notice.is_faculty or notice.is_instructor or
                 notice.is_past_employee or notice.is_retiree or
-                notice.is_staff_employee or notice.is_stud_employee):
+                notice.is_staff_employee or notice.is_stud_employee or
+                notice.is_intl_stud):
```
Comment (fanglinfang) [inline myuw/dao/myuw_notice.py]: cause #1.

File: myuw/dao/myuw_notice.py [hunk 3]
```diff
@@ -1,60 +1,65 @@
 from myuw.dao.affiliation import get_all_affiliations
 from myuw.models.myuw_notice import MyuwNotice
-from myuw.dao.term import get_comparison_datetime
+from myuw.dao.term import get_comparison_datetime_with_tz
 
 
 def get_myuw_notices_for_user(request):
-    date = get_comparison_datetime(request)
+    date = get_comparison_datetime_with_tz(request)
```
Comment (fanglinfang) [inline myuw/dao/myuw_notice.py]: For RuntimeWarning: DateTimeField MyuwNotice.end received a naive datetime (2018-09-07 06:13:00) while time zone support is active

File: myuw/test/views/notice_admin.py [hunk 4]
```diff
@@ -120,25 +123,30 @@ def test_html_content(self):
...
+        notice_context['content'] = "Bar"
+        notice_context['title'] = 'Edited'
+
+        entries = MyuwNotice.objects.all()
+        self.assertEqual(len(entries), 1)
 
         request = rf.post('', notice_context)
-        _save_notice(request, {}, 1)
+        self.assertTrue(_save_notice(request, {},
+                                     notice_id=entries[0].id))
```
Comment (fanglinfang) [inline myuw/test/views/notice_admin.py]: When running all the tests together, the id is not 1. This query will fail.

File: myuw/dao/myuw_notice.py (hunk 1 above)
Comment (devights) [reply myuw/dao/myuw_notice.py]: Won't this break the campus based notice filtering?

**Comment by fanglinfang** [comment]
No, otherwise if none of the affiliation matched but campus did, the notice will be displayed.

File: myuw/test/dao/myuw_notice.py [hunk 5]
```diff
@@ -74,38 +81,60 @@ def test_affil(self):
...
+                            notice_category="Student",
+                            start=get_datetime_with_tz(2018, 6, 8, 10),
+                            end=get_datetime_with_tz(2018, 6, 20, 10),
+                            is_seattle=True,
+                            is_intl_stud=True)
+        notice.save()
         request = get_request_with_date("2018-06-09")
         get_request_with_user('javerage', request)
         notices = get_myuw_notices_for_user(request)
         self.assertEqual(len(notices), 1)
```
Comment (fanglinfang) [inline myuw/test/dao/myuw_notice.py]: Stephen, without the change on line 48 above, this assertion on line 108 will fail due to the lenngth being 2.

**Pull request merged**